controllers/spacebindingcleanup/spacebinding_cleanup_controller.go

```diff
@@ -0,0 +1,95 @@
+package spacebindingcleanup
+
+import (
+	"context"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/go-logr/logr"
+	errs "github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	onlyForDeletion := builder.WithPredicates(OnlyDeleteAndGenericPredicate{})
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&toolchainv1alpha1.SpaceBinding{}).
+		Watches(&source.Kind{Type: &toolchainv1alpha1.Space{}},
+			handler.EnqueueRequestsFromMapFunc(MapToSpaceBindingByBoundObjectName(r.Client, toolchainv1alpha1.SpaceBindingSpaceLabelKey)),
+			onlyForDeletion).
+		Watches(&source.Kind{Type: &toolchainv1alpha1.MasterUserRecord{}},
+			handler.EnqueueRequestsFromMapFunc(MapToSpaceBindingByBoundObjectName(r.Client, toolchainv1alpha1.SpaceBindingMasterUserRecordLabelKey)),
+			onlyForDeletion).
+		Complete(r)
+}
+
+// Reconciler reconciles a SpaceBinding object
+type Reconciler struct {
+	client.Client
+	Scheme    *runtime.Scheme
+	Namespace string
+}
+
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spacebindings,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spacebindings/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spacebindings/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithName("cleanup")
```

Comment:
yes, there is enough information already. The function `FromContext` already populates the log with needed information - it adds:
```
"logger":"controller-runtime.manager.controller.<CRD-name>"
```
so in this case:
```
"logger":"controller-runtime.manager.controller.spacebindings"
```
however this can be confusing as this is not a normal SpaceBinding controller, but cleanup controller. The `WithName("cleanup")` adds another part to the logger name:
```
"logger":"controller-runtime.manager.controller.spacebindings.cleanup"
```

Reply:
Great, thanks for the explanation!